lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -155,35 +175,42 @@ class TraceElements extends Gatherer {
 
     const lcpNodeId = TraceElements.getNodeIDFromTraceEvent(largestContentfulPaintEvt);
     const clsNodeData = TraceElements.getTopLayoutShiftElements(mainThreadEvents);
-    if (lcpNodeId) {
-      backendNodeData.push({nodeId: lcpNodeId});
-    }
-    backendNodeData.push(...clsNodeData);
+    const animatedElementData = TraceElements.getAnimatedElements(mainThreadEvents);
 
-    const traceElements = [];
-    for (let i = 0; i < backendNodeData.length; i++) {
-      const backendNodeId = backendNodeData[i].nodeId;
-      const metricName =
-        lcpNodeId === backendNodeId ? 'largest-contentful-paint' : 'cumulative-layout-shift';
-      const objectId = await driver.resolveNodeIdToObjectId(backendNodeId);
-      if (!objectId) continue;
-      const response = await driver.sendCommand('Runtime.callFunctionOn', {
-        objectId,
-        functionDeclaration: `function () {
-          ${setAttributeMarker.toString()};
-          ${pageFunctions.getNodePathString};
-          ${pageFunctions.getNodeSelectorString};
-          ${pageFunctions.getNodeLabelString};
-          ${pageFunctions.getOuterHTMLSnippetString};
-          ${pageFunctions.getBoundingClientRectString};
-          return setAttributeMarker.call(this, '${metricName}');
-        }`,
-        returnByValue: true,
-        awaitPromise: true,
-      });
+    const backendNodeDataMap = new Map([
+      ['largest-contentful-paint', lcpNodeId ? [{nodeId: lcpNodeId}] : []],
+      ['cumulative-layout-shift', [...clsNodeData]],
+      ['CLS/non-composited-animations', [...animatedElementData]],
+    ]);
 
-      if (response && response.result && response.result.value) {
-        traceElements.push({...response.result.value, score: backendNodeData[i].score});
+    const traceElements = [];
+    for (const [metricName, backendNodeData] of backendNodeDataMap) {
```

Comment:
@paulirish were you onboard for renaming `metricName` to something else?

`traceEventType: 'largest-contentful-paint'|'layout-shift'|'animation'` perhaps?

Reply:
+1 to renaming. I don't believe `metricName` makes sense any longer.